tests/utils/test_task_group.py

```diff
@@ -0,0 +1,529 @@
+        with TaskGroup("group234", prefix_group_id=False) as group234:
+            task2 = DummyOperator(task_id="task2")
+
+            with TaskGroup("group34") as group34:
+                task3 = DummyOperator(task_id="task3")
+
+                with TaskGroup("group4", prefix_group_id=False) as group4:
+                    task4 = DummyOperator(task_id="task4")
+
+        task5 = DummyOperator(task_id="task5")
+        task1 >> group234
+        group34 >> task5
+
+    assert task2.task_id == "task2"
+    assert group34.group_id == "group34"
+    assert task3.task_id == "group34.task3"
+    assert group4.group_id == "group34.group4"
+    assert task4.task_id == "task4"
+    assert task5.task_id == "task5"
+    assert group234.get_child_by_label("task2") == task2
+    assert group234.get_child_by_label("group34") == group34
+    assert group4.get_child_by_label("task4") == task4
+
+    assert extract_node_id(task_group_to_dict(dag.task_group), include_label=True) == {
+        'id': None,
+        'label': None,
+        'children': [
+            {
+                'id': 'group234',
+                'label': 'group234',
+                'children': [
+                    {
+                        'id': 'group34',
+                        'label': 'group34',
+                        'children': [
+                            {
+                                'id': 'group34.group4',
+                                'label': 'group4',
+                                'children': [{'id': 'task4', 'label': 'task4'}],
+                            },
+                            {'id': 'group34.task3', 'label': 'task3'},
+                            {'id': 'group34.downstream_join_id', 'label': ''},
+                        ],
+                    },
+                    {'id': 'task2', 'label': 'task2'},
+                    {'id': 'group234.upstream_join_id', 'label': ''},
+                ],
+            },
+            {'id': 'task1', 'label': 'task1'},
+            {'id': 'task5', 'label': 'task5'},
+        ],
+    }
+
+
+def test_build_task_group_with_task_decorator():
+    """
+    Test that TaskGroup can be used with the @task decorator.
+    """
+    from airflow.operators.python import task
+
+    @task
+    def task_1():
+        print("task_1")
+
+    @task
+    def task_2():
+        return "task_2"
+
+    @task
+    def task_3():
+        return "task_3"
+
+    @task
+    def task_4(task_2_output, task_3_output):
+        print(task_2_output, task_3_output)
+
+    @task
+    def task_5():
+        print("task_5")
+
+    execution_date = pendulum.parse("20200101")
+    with DAG("test_build_task_group_with_task_decorator", start_date=execution_date) as dag:
+        tsk_1 = task_1()
+
+        with TaskGroup("group234") as group234:
+            tsk_2 = task_2()
+            tsk_3 = task_3()
+            tsk_4 = task_4(tsk_2, tsk_3)
+
+        tsk_5 = task_5()
+
+        tsk_1 >> group234 >> tsk_5
```

Comment:
@turbaszek thanks for #10827. Now its merged, i've updated this test to this more natural syntax.